File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.py
class Solution(object):
    def consecutiveSetBits(self, n):
        """
        :type n: int
        :rtype: bool
        """
        cnt, p = 0, -1
        while n:
            r = 1 if n & 1 else 0
            if r == 1 and r == p:
                cnt += 1
            
            p = r
            n >>= 1
        
        return cnt == 1

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.cs
public class Solution {
    public bool ConsecutiveSetBits(int n) {
        int c = 0, p = -1;
        for (; n > 0; n >>= 1) {
            int r = ((n & 1) != 0) ? 1 : 0;
            if (r == 1 && r == p)
                c++;
            p = r;
        }
        return c == 1;
    }
}